### PR TITLE
Add ActionableMessageTokenValidator class to simplify the token validation task. 

### PR DESCRIPTION
**Pull request by wowong**

### CODE REVIEW

File: ActionableMessageTokenValidator.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+// 
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// 
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
```
Comment (markenki) [inline ActionableMessageTokenValidator.js]: Do we need this?

File: ActionableMessageTokenValidator.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        var openIdMetadata = new oid.OpenIdMetadata(O365_OPENID_METADATA_URL)
+        
+        openIdMetadata.getKey(decoded.header.kid, key => {
+            var result = new ActionableMessageTokenValidationResult();
+            
+            if (key) {
+                try {
+                    jwt.verify(token, key, verifyOptions);
+                    
+                    if (decoded.payload.appid != O365_APP_ID) {
```
Comment (markenki) [inline ActionableMessageTokenValidator.js]: Needs to be case insensitive.

File: app.js [hunk 3]
```diff
@@ -45,55 +44,44 @@ app.post('/api/expense', function (req, res) {
...
+            "https://api.contoso.com",
+            function (err, result) {
+                if (err) {
+                    console.error('error: ' + err.message);
+                    res.status(401);
+                    res.end();
+                } else {
+                    // We have a valid token. We will verify the sender and the action performer. 
+                    // You should replace the code below with your own validation logic.
+                    // In this example, we verify that the email is sent by Contoso LOB system
```
Comment (markenki) [inline app.js]: with email address lob@contoso.com

File: app.js [hunk 4]
```diff
@@ -45,55 +44,44 @@ app.post('/api/expense', function (req, res) {
...
+                    // and the action performer has to be someone with @contoso.com email.
```
Comment (markenki) [inline app.js]: and the action performer is someone with a @contoso.com email address.

File: ActionableMessageTokenValidator.js (hunk 1 above)
Comment (wowong) [reply ActionableMessageTokenValidator.js]: Fixed.

File: ActionableMessageTokenValidator.js (hunk 2 above)
Comment (wowong) [reply ActionableMessageTokenValidator.js]: Fixed.

File: app.js (hunk 3 above)
Comment (wowong) [reply app.js]: Fixed.

File: app.js (hunk 4 above)
Comment (wowong) [reply app.js]: Fixed.

**Pull request merged**